### ISSUE TITLE
Boolean positional parameters in sql query doesn't work (2.1 regression)

### ISSUE THREAD

**Issue by xvik** [orientechnologies/orientdb#4224]
```java
class User {
    private String username;
    private boolean active;
    // skip: id and version mapped, getters setters present
}
```
```sql
select from User where username = ? and active = ? limit 1
```
This query works for 2.0.10, but return empty result for 2.1-rc3.

Query executed with object connection (OObjectDatabaseTx) using OSQLSynchQuery

Named parameters work correctly:
```sql
select from User where username = :name and active = :active limit 1
```
So the problem is only related to positional parameters.

[Test for reproducing the problem](https://github.com/xvik/guice-persist-orient/blob/master/src/test/groovy/ru/vyarus/guice/persist/orient/study/boolparam/BoolParamTest.groovy) (if you change project orient dependency from 2.0.10 to 2.1-rc3 it will fail)

**Comment by xvik** [comment]
Problem appear because `com.orientechnologies.orient.core.sql.parser.OInputParameter#toParsedTree` doesn't support Boolean.

##### What happens

When we execute this query 
```sql
select from User where username = ? and active = ? limit 1
```
with parameters 'test' and true

During pre-parse here
`com.orientechnologies.orient.core.sql.OCommandExecutorSQLAbstract#preParse` it substitutes some variables in query directly. And because `com.orientechnologies.orient.core.sql.parser.OInputParameter#toParsedTree` knows string, but not boolean we have this query for execution:

```sql
select from User where username = 'test' and active = ? limit 1
```
Now when it executes pre-processed query it see one(!) positional argument and *takes first argument from parameters* (but actual value is second argument)! 


##### When it works

When we use named parameters

```sql
select from User where username = :name and active = :active limit 1
```
During pre-parse user parameter is again replaced directly and ':active' stays as named
```sql
select from User where username = 'test' and active = :active limit 1
```
But now it takes parameter by name and that's why it works properly.

Also, if we switch conditions
```sql
select from User where active = ? and username = ? limit 1
```
It will work (because its always gets first parameter and luckilly its correct in this case)

**Comment by luigidellaquila** [comment]
Hi @xvik 

It's already fixed in 2.1.x branch and in latest hotfix. I just added a test case for this.
Closing, please reopen if the problem arises again
